controllers/webhook/webhook.go

```diff
@@ -135,35 +138,43 @@ type OpenDataHubMutatingWebhook struct {
 	Decoder *admission.Decoder
 }
 
-func (w *OpenDataHubMutatingWebhook) SetupWithManager(mgr ctrl.Manager) {
+func (m *OpenDataHubMutatingWebhook) SetupMutateWebhookWithManager(mgr ctrl.Manager) {
 	hookServer := mgr.GetWebhookServer()
 	odhWebhook := &webhook.Admission{
-		Handler: w,
+		Handler: m,
 	}
 	hookServer.Register("/mutate-opendatahub-io-v1", odhWebhook)
 }
 
-func (w *OpenDataHubMutatingWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
-	var resp admission.Response
-	resp.Allowed = true // initialize Allowed to be true in case Operation falls into "default" case
-
+func (m *OpenDataHubMutatingWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
 	dsc := &dscv1.DataScienceCluster{}
-	err := w.Decoder.Decode(req, dsc)
+	err := m.Decoder.Decode(req, dsc)
 	if err != nil {
 		return admission.Denied(fmt.Sprintf("failed to decode request body: %s", err))
 	}
-	resp = w.setDSCDefaults(ctx, dsc)
-	return resp
+
+	if err := m.Default(ctx, dsc); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// convert to []byte
+	dscBytes, err := json.Marshal(dsc)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	// use PatchResponseFromRaw to ensure dsc CR is updated with defaults
+	return admission.PatchResponseFromRaw(req.Object.Raw, dscBytes)
```

Comment:
so finally this is the actual change :) But still, could you put some reasoning behind?